tests/src/python/test_qgstextrenderer.py

```diff
@@ -2285,6 +2285,16 @@ def testDrawBufferSizePixels(self):
         format.buffer().setSizeUnit(QgsUnitTypes.RenderPixels)
         assert self.checkRender(format, 'text_buffer_pixels', QgsTextRenderer.Buffer, text=['test'])
 
+    def testDrawBufferSizePercentage(self):
```

Comment:
Might be worth testing this % unit type against a data-defined font size property if the checkRender can handle this.

Reply:
Data defined properties with text format doesn't really work that way -- you call updateDataDefinedProperties() on the text format itself and all its properties are updated in place

Reply:
@nyalldawson , roger that.